src/plugins/mdraid.c

```diff
@@ -993,9 +991,8 @@ BDMDExamineData* bd_md_examine (const gchar *device, GError **error) {
         return NULL;
     }
 
-    ret = get_examine_data_from_table (table, TRUE, error);
+    ret = get_examine_data_from_table (table, TRUE);
     if (!ret)
-        /* error is already populated */
```

Comment:
Hmm, wait - if `NULL` is a valid return value, it should be noted in the doc strings. However I would advise to keep the usual semantics of _return NULL means error is set_. Breaking the API is one option we can afford.

Reply:
`get_examine_data_from_table` never set the error and really couldn't return `NULL` anyway so the this check was always questionable. But I've added error here in case we change `get_examine_data_from_table` in the future.